internal/plugin/platform.go

```diff
@@ -436,7 +437,13 @@ func (s *platformServer) Deploy(
 		return nil, err
 	}
 
-	result := &proto.Deploy_Resp{Result: encoded}
+	result := &proto.Deploy_Resp{Result: encoded, Deployment: &proto.Deploy{}}
+
+	switch raw.(type) {
+	case component.DeploymentWithUrl:
+		result.Deployment.Url = raw.(component.DeploymentWithUrl).URL()
+	}
```

Comment:
One other quick thing, could you either use assignment in the switch cond `r := raw.(type)`, use the condition form of the type check, so that we don't have to do the panic-possible check again on 444.